Update current examine documentation (#5195)
umbraco/UmbracoDocs: 11/umbraco-cms/reference/searching/examine/indexing.md

            _productIndexValueSetBuilder = productIndexValueSetBuilder;
            RegisterIndex("ProductIndex");
        }

        protected override void PopulateIndexes(IReadOnlyList<IIndex> indexes)
        {
            foreach (IIndex index in indexes)
            {
                IContent[] roots = _contentService.GetRootContent().ToArray();
                index.IndexItems(_productIndexValueSetBuilder.GetValueSets(roots));

                foreach (IContent root in roots)
                {
                    const int pageSize = 10000;
                    var pageIndex = 0;
                    IContent[] descendants;
                    do
                    {
                        descendants = _contentService.GetPagedDescendants(root.Id, pageIndex, pageSize, out _).ToArray();
                        IEnumerable<ValueSet> valueSets = _productIndexValueSetBuilder.GetValueSets(descendants);
                        index.IndexItems(valueSets);

                        pageIndex++;
                    }
                    while (descendants.Length == pageSize);
                }
            }
        }
    }
}
```
{% hint style="info" %}

> Note here that this is just an example of how you could do indexing, this means that we're indexing all content, also unpublished & trashed content.

Reviewer: ```suggestion
Note that this is just an example of how you could do indexing. In this sample we're indexing all content, also unpublished content.
```

Since we're only fetching descendants of content roots, we can't ever be indexing trashed content here?

> If you want to filter those out, you'd have to make your own logic for that.

Reviewer: ```suggestion
If you want to filter those out, you will have to create your own logic for that.
```

> You could as an example look at published status in the `ProductIndexValueSetBuilder.GetValueSets()` method and filter it there.

Reviewer: Not entirely correct; the published status can change for sub pages, leaving potentially unroutable published content in the index. Probably better to link to the `IndexPublishedContent` method of `ContentIndexPopulator`, which performs a structural filtering of published content below unpublished.

Author: I have tried rewriting this 👍

}
```

{% hint style="info" %}

> The order of registration may seem arbitrary, but it's important so register your index, before `ConfigureOptions`.

Reviewer: ```suggestion
The order of these registration matter. It is important to register your index with `AddExamineLuceneIndex` before calling `ConfigureOptions`.
```
